### PR TITLE
Add LGTM configuration file for LGTM tool

### PR DESCRIPTION
**Pull request by zbud-msft**
Added lgtm.yml as a requirement for LGTM tool to establish automated code review using Go linter. 
Default workspace for LGTM is /opt/src, so for the purposes of the LGTM tool build, we can change the location of the workspace to /opt/src instead of parent director /opt.

### CODE REVIEW

**Comment by renukamanavalan** [comment]
Can you please provide the steps taken to validate this change ?
Has it been tried with a .go that has no bug and as well one with bugs ?

**Comment by zbud-msft** [comment]
> Can you please provide the steps taken to validate this change ? Has it been tried with a .go that has no bug and as well one with bugs ?

Yes, attaching the LGTM tool log: https://lgtm.com/logs/00d47310d3af018d3beb998f24e3df418c79cde7/lang:go

1. I see that the current language analysis for sonic-telemetry is Python and there is no lgtm configuration file, so I added a lgtm configuration file that specified to use go as the testing language.
2. Since go.mod requires sonic-mgmt-common, and specifies that it should be located in the parent directory, LGTM cannot install the sonic-mgmt-common repository in it's parent directory as that is not allowed. Which is why for the purposes of LGTM, I change that the sonic-mgmt-common should be in LGTM's current working directory which is /opt/src
3. We can verify that the build passes as we see it in the logs. In the analysis tab, we can also see the current warnings

File: lgtm.yml
Comment (qiluo-msft) [inline lgtm.yml:4]: Instead of modify Makefile directly, is it possible to override the `MGMT_COMMON_DIR` variable?
I see "LGTM_SRC is /opt/src" in lgtm log.
LGTM support environment variable https://lgtm.com/help/lgtm/customizing-code-extraction

File: lgtm.yml
Comment (qiluo-msft) [inline lgtm.yml:6]: Is it same as just deleting this line?

File: lgtm.yml
Comment (zbud-msft) [reply lgtm.yml]: After some research and trial/error, changes to Makefile are not needed to have the build succeed with Go language

File: lgtm.yml
Comment (zbud-msft) [reply lgtm.yml]: The changes to go.mod are needed to simply tell LGTM that the sonic-mgmt-common directory is located in the current director (./sonic-mgmt-common) and not in the parent directory (../sonic-mgmt-common)

**Review by qiluo-msft — APPROVED** [review]

**Comment by qiluo-msft** [comment]
I believe Go language LGTM will be enabled after this PR merged.

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**
